roles/kubernetes-apps/container_engine_accelerator/nvidia_gpu/templates/xenial-nvidia-driver-install-daemonset.yml.j2

```diff
@@ -0,0 +1,93 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: nvidia-driver-installer
+  namespace: kube-system
+spec:
+  template:
+    metadata:
+      labels:
+        name: nvidia-driver-installer
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: "nvidia.com/gpu"
+                operator: Exists
+      tolerations:
+      - key: "nvidia.com/gpu"
+        effect: "NoSchedule"
+        operator: "Exists"
+      hostNetwork: true
+      hostPID: true
+      volumes:
+      - name: dev
+        hostPath:
+          path: /dev
+      - name: nvidia-install-dir-host
+        hostPath:
+          path: /home/kubernetes/bin/nvidia
+      - name: root-mount
+        hostPath:
+          path: /
+      - name: modules
+        hostPath:
+          path: /lib/modules
+      initContainers:
+      - image: atzedevries/xenial-nouveau-unloader:1
```

Comment:
There are 3 docker images in play. A pause image, a installer image and a nouveau unloader. The nouveau unloader is not in the GCP since nouveau is not available in GCP (and they are not going to add it). GCP systems are not same as most of 'our' installs. 

By default in ubuntu/centos installed nouveau is loaded. Unloading before install is required. We can ofcourse leave this to the end user. By adding this simple extra init container (with image `xenial-nouveau-unloader`) we can control this step and make sure the nvidia driver is correctly loaded. The nvidia driver can only be loaded by the `gcr.io/google-containers/ubuntu-nvidia-driver-installer` container. 

My advice would be to keep it like this since it add more control to a succesfull install and the difference is only a extra initContianer, so keeping up with GCP source should be simple.  

An other difference is:
```YAML
- matchExpressions:
              - key: cloud.google.com/gke-accelerator
                operator: Exists
```
This is very GCP specific, so i've changed it to
```YAML
- matchExpressions:
              - key: "nvidia.com/gpu"
                operator: Exists
```
To make it more uniform. But this is just a style choice. 

I'm not fully aware of how kupespray hosts self developed images, but for me it makes sense if kubespray hosts this image. I can deliver the `Dockerfile`, or we can rewrite it into a default xenial image with a `command` and some `args` which does the trick.